src/HealthCheckServer.ts

```diff
@@ -64,11 +62,8 @@ const defaultHealthHandler = (healthCheck: any): Lifecycle.Method => {
  * @description the failure handler for Hapi. We put this here to make it more testable
  *
  */
-const failAction = async (_request: Hapi.Request, _handler: Hapi.ResponseToolkit, err?: Error) => {
-  if (!err) {
-    throw Boom.boomify(new Error(`Unknown Server Error`))
-  }
-  throw Boom.boomify(err)
+const failAction = async (_request: Hapi.Request, _handler: Hapi.ResponseToolkit, err?: Error): Promise<void> => {
+  throw ErrorHandler.Factory.reformatFSPIOPError(err)
```

Comment:
If the failAction has to catch joi validation requests from http server then I could suggest using the createFSPIOPErrorFromJoi function